osv/ecosystems/cran.py

```diff
@@ -0,0 +1,69 @@
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""CRAN helpers."""
+
+import packaging.version
+import requests
+
+from . import config
+from .helper_base import Ecosystem, EnumerateError
+
+class CRAN(Ecosystem):
+  """CRAN ecosystem helpers."""
+
+  # Use the Posit Public Package Manager API to pull both the current and archived versions for a specific package
+  # since CRAN doesn't natively support this functionality.
+  _API_PACKAGE_URL_POSIT_CRAN = 'https://packagemanager.posit.co/__api__/repos/2/packages/{package}'
+
+  def sort_key(self, version):
+    """Sort key."""
+    return packaging.version.parse(version)
```

Comment:
Is Python's versioning scheme suitable for CRAN?
Looking at [this page](https://r-pkgs.org/lifecycle.html#sec-lifecycle-version-number), R seems to treat both `.` and `-` as the same separator, which Python does not.

Though, I don't know if Posit will show versions using `-`'s?

Also, could you add a link to that R version number (or a better page?) in a comment for reference.

Reply:
`semver` is probably worse in this case, since it will error on `w.x.y.z`.

Reply:
I added some reference links, and additional cases for the typical `X.Y-Z` I see in CRAN. Looks like the `packaging.version.parse(...)` method handles these well and should work moving forward.

Reply:
`packaging.version.parse(...)` has some odd behaviour surrounding `-` that would be wrong for R:

```python
from packaging.version import parse
# All these evaluate as True
parse('1.0') < parse('1.0-0')
parse('1.0.0') < parse('1.0-0') 
parse('1.0-0') > parse('1.0-0.0') 
parse('1.0-0') > parse('1.0-0.1')
parse('1.0-0') > parse('1.0-1.0')
```

I think parsing `version.replace('-', '.')` will give the correct comparisons, though.